src/item.cpp

```diff
@@ -5015,6 +5015,11 @@ bool item::fill_with( item &liquid, std::string &err, bool allow_bucket )
     return true;
 }
 
+void item::set_countdown( int num_turns )
+{
+    charges = num_turns;
```

Comment:
You could a `debumsg` for `ammo_type() != "NULL"` that would catch future obvious abuse of that function. Similarly it could be worth checking for negative `num_turns` as `charges` doesn't become negative anywhere else.